### PR TITLE
[Rewards] Updates generic linking error text.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes https://github.com/brave/brave-browser/issues/33955. Only addresses the text change, but doesn't wire up the http status code.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_rewards/resources/page/components/provider_redirect_modal.tsx [hunk 1]
```diff
@@ -187,12 +180,13 @@ export function ProviderRedirectModal () {
       return (
         <ModalRedirect
           id={'redirect-modal-error'}
-          errorText={[getString('redirectModalError')]}
-          buttonText={getString('processingRequestButton')}
-          titleText={getString('processingRequest')}
+          errorText={[getString('redirectModalUnableToCompleteRequest')]}
+          errorTextLink={'https://community.brave.com'}
```
Comment (zenparsing) [inline components/brave_rewards/resources/page/components/provider_redirect_modal.tsx:184]: Is this one already in `shared/lib/rewards_urls.ts`?

**Review by zenparsing — APPROVED** [review]

File: components/brave_rewards/resources/page/components/provider_redirect_modal.tsx (hunk 1 above)
Comment (mkarolin) [reply components/brave_rewards/resources/page/components/provider_redirect_modal.tsx:184]: Yes, it is. I'll replace them through the file.

Comment (mkarolin) [reply components/brave_rewards/resources/page/components/provider_redirect_modal.tsx]: https://github.com/brave/brave-core/pull/21863/commits/485dd84d7a96994e42f712a486bcf1929a7c4c47

**Review by bsclifton — APPROVED** [review]
text changes LGTM!

**Pull request merged**